web/client/plugins/MousePosition.jsx

```diff
@@ -64,11 +64,22 @@ const MousePositionButton = connect((state) => ({
 
 const MousePositionComponent = require('../components/mapcontrols/mouseposition/MousePosition');
 
-
 class MousePosition extends React.Component {
+
     render() {
+
+        const degreesTemplateComponent = ((degreesTemplateStr) => {
+            switch( degreesTemplateStr ) {
```

Comment:
move this function outside render() and remove application of parameter
like so
```
getDegreesTemplateComponent = (degreesTemplateStr) => {
        switch (degreesTemplateStr) {
            case "MousePositionLabelDD" : return require('../components/mapcontrols/mouseposition/MousePositionLabelDD');
            case "MousePositionLabelDM" : return require('../components/mapcontrols/mouseposition/MousePositionLabelDM');
            case "MousePositionLabelDMS" : return require('../components/mapcontrols/mouseposition/MousePositionLabelDMS');
            case "MousePositionLabelDMSNW" : return require('../components/mapcontrols/mouseposition/MousePositionLabelDMSNW');
            default: return require('../components/mapcontrols/mouseposition/MousePositionLabelDMS');
        }
    };
```

inside render
```
let degreesTemplateComponent = this.getDegreesTemplateComponent(this.props.degreesTemplateStr);
        return (
            <MousePositionComponent toggle={<MousePositionButton/>} degreesTemplate={degreesTemplateComponent} {...this.props}/>
        );
```